01/components/Table.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { TableHeader, TableRow, TableFooter } from './Table/TableOrganizer';
+
+class Table extends React.Component {
+	render() {
+		const { data } = this.props;
+		//jak to sie dzieje ze nigdzie nie detchuje data a w consoli juz mam pliki sprasowane z jsona?
+		console.log(data);
+
+		// Zwróć uwagę na przekazywanie danych z rodzica do dziecka. ## Nie przekazuj wszystkich danych, lecz tylko te, które będą wykorzystane w dziecku ##.
+
+		// czy to warto glowny komponent Table zasmiecac jak mozemy przeiterowac sobie po calym obiekcie wewnatrz TableFooter zamiast podawac tylko wybrane?
+		const priceQuantity = data.map(item => ({
+			price: item.price,
+			quantity: item.quantity,
+		}));
```

Comment:
Osobiście wydaje mi się, że nie jest to zaśmiecanie (w szczególności jeśli użyjemy do tego funkcji/metody) tylko określanie co potrzebujemy.

Podszedłbym do tego tak jak do funkcji. Czy w parametrach przekazujesz wszystkie dane czy tylko te niezbędne do jej działania?

PS. Czy w tym miejscu nie możesz po prostu przekazać już sumy, którą chcesz wyświetlić?

Reply:
da sie wrzucic totalSum ktore odpowiada za wyliczenie calkowitej ceny do obiektu priceQuantity? czy w tej postaci jest dobrze?

Reply:
Chodziło mi o zrobienie np.; `cons totalPrice = data.reduce((acc, curr) => (curr.price * curr.quantity) + acc, 0)` i przekazanie `totalPrice` do `fotoer`